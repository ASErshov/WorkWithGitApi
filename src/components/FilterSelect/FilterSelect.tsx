import * as React from 'react'
import { MenuItem, Select, InputLabel } from '@material-ui/core'

export type FilterSelectProps ={
    handleSelect(value:string|number): void,
    lable: string,
    selected?: string|number,
    items: Array<{
        name: string,
        key: string|number
    }>
}

const FilterSelect: React.FC<FilterSelectProps> = ({handleSelect, lable, items, selected}): React.ReactElement =>{

    const handleChange = (e:React.ChangeEvent<{ value: unknown }>)=>{
        handleSelect(e.target.value as string|number)
    }

    return (
        <div>
            <InputLabel id="label">{lable}</InputLabel>
            <Select
            labelId="label"
            id="demo-simple-select"
            value={selected}
            onChange={handleChange}
            style ={{width: '100%'}}
            >
                <MenuItem value={0}>
                    <em>None</em>
                </MenuItem>
                {items.map(item =>
                <MenuItem key={`${lable}_filter_${item.key}`} value={item.key}>{item.name}</MenuItem>
                )}

            </Select>
        </div>
    )
}

export default FilterSelect