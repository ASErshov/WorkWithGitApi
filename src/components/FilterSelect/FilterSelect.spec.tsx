import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import styles from './Button.module.css'

import FilterSelect from './index'

describe('<FilterSelect />', () => {
  const onClick = jest.fn()

  const Component = (
    <FilterSelect
      handleSelect = {onClick}
      lable = 'Фильтр по лицензии'
      items = {[{name: '1', key: 1},{name: '2', key: 2},{name: '3', key: 3}]}
      selected = {1}
    />
  )

  it('Snapshot is correct', () => {
    const wrapper = renderer.create(Component).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('Passed props is valid', () => {
    const wrapper = mount(Component)

    expect(wrapper.props().lable).toEqual('Фильтр по лицензии')
    expect(wrapper.props().items).toEqual([{name: '1', key: 1},{name: '2', key: 2},{name: '3', key: 3}])
    expect(wrapper.props().selected).toEqual(1)
  })

  it('JSX has been rendered is valid', () => {
    const wrapper = mount(Component)

    expect(wrapper.find('label').text()).toEqual('Фильтр по лицензии')
    expect(wrapper.find('div.MuiInput-input').text()).toEqual('1')
  })
})
