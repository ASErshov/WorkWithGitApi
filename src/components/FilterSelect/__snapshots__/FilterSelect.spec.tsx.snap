// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSelect /> Snapshot is correct 1`] = `
<div>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
    id="label"
  >
    Фильтр по лицензии
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline"
    onClick={[Function]}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="label demo-simple-select"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="demo-simple-select"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      1
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      tabIndex={-1}
      value={1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
